### ISSUE TITLE
Am not allowed to specify password when running git phlow auth (unresolved)

### ISSUE THREAD

**Issue by RandomSort** [Praqma/git-phlow#274]
**Scenario**
When running `git phlow auth` I get the following result:
```
$ git phlow auth
Enter credentials for jira
username: test
password:
Could not verify credentials
```
This might of course be valuable, but rather than waiting for input on the password field it skips immediately further.

I can demonstrate this on hangout if neccesary.

**Environment:**
git bash
Windows 10
Atlassian JIRA Project Management Software (v7.5.0#75005-sha1:fd8c849) 
```
**config:**
$ head ~/.gitconfig
[phlow]
remote                 = origin
service                = jira
integration-branch     = master
issue-api              = https://jira.redacted.com
issue-web              = https://jira.redacted.com
delivery-branch-prefix = ready
```

**Comment by RandomSort** [comment]
Anecdotally after having done this a few times I was forced to use a CAPTCHA before logging in through the webinterface. This is not a common occurence.

**Comment by groenborg** [comment]
Further description of this issues would be helpful @RandomSort. Is the issue that it rejects your password?

**Comment by RandomSort** [comment]
As I wrote in the issue I'm not allowed to provide my password, it moves on from that prompt before I can do that.

This is very anecdotally, but it feels like some request is made to Jira before I can write my password and this fails. But this is *very* anecdotally.

**Comment by buep** [comment]
what do you mean by _anecdotally_  was it a one of problem, or is it easy to reproduce?

**Comment by RandomSort** [comment]
This happens repeatedly without exception on the machine I'm on at the moment. The anecdote was the feel about the request to Jira. But that I do not have any evidence to support.

Every single time I run `git phlow auth` it fails to allow me to provide a password.

**Comment by RandomSort** [comment]
I tried in powershell and it works here. 
Git bash support would of course be nice since it tends to be the CLI of choice for many Git users on Windows. But perhaps this just went from bug to 'nice-to-have'.

Sorry for the disturbance.

**Comment by buep** [comment]
Git bash is on our white-list and should be supported. So if it doesn't work it a bug.

I'm unsure if we can reproduce this, so we need to detect what is different in your setup.

We will try basic setup first.

**Comment by groenborg** [comment]
If I had to make a guess, then there is some stuff happening with line endings.. Which is very interesting since we use go internal for that

**Comment by groenborg** [comment]
We did not run into this problem when we tested it back at the code cafe it should have been solved In #158

**Comment by MadsNielsen** [comment]
I got curios about this, maybe you can comment @groenborg on this. Is there a reason you're not passing std.in when reading the password. The ReadInput method is called twice but only uses the std.in for the username?

https://github.com/Praqma/git-phlow/blob/master/phlow/auth.go#L63-L67

**Comment by groenborg** [comment]
@MadsNielsen They both uses stdin, the one from the password comes from syscall. It is made this way to hide the password when you type it. It should not cause an issue. but I might test it out

**Comment by groenborg** [comment]
@RandomSort I have verified your issue

**Comment by groenborg** [comment]
The syscall does not make the terminal wait for input on git bash. First though was that the stdin was not empty and caught the linebreak from the previous command but that should not be the issue. There is some terminal stuff I am not familiar with, this stackoverflow question could provide some answers https://stackoverflow.com/questions/22744443/check-if-there-is-something-to-read-on-stdin-in-golang

**Comment by groenborg** [comment]
This issue seems to apper with emulated terminals on windows, like git bash and Cmder. They both won't read from stdin when it prompts for password. I am still investigating why, but it could have something to do with tty and piping

**Comment by MadsNielsen** [comment]
Maybe an experiment with: https://github.com/howeyc/gopass

**Comment by groenborg** [comment]
I have searched the net thin for the answer to this, but it seems that supporting git-bash is no easy solution. I have tried various packages and forum threads but nothing seems to work

**Comment by buep** [comment]
Interesting... but when I run my commands myself that asks for password, e.g. cloning over http I work in git bash, so isn't our "layer" what that is that interacts with git bash that is causing the problem?

**Comment by groenborg** [comment]
Yes it is our layer that fails in git bash.. but I am at a point where I think that the issue is go itself. I find it funny that not a single package can successfully silent the tty when doing an input on stdin

**Comment by buep** [comment]
We agreed @groenborg  will write to all the open source libraries he tried and ask why it doesn't work with emulated bash terminals like Git Bash or Cmd'er as we use a library to hide and capture passwords.
Simon tried 3-4 libs and will write them all... link the question or issue into this issue so we can all follow up on it.

After that we will try to downgrade to go 1.8 from currently 1.9 as we upgrade this recently and we think it might have been working before.

@groenborg  will also ask the go community more generally in this matter. Link to here as well.

After that we will "park" this issue shortly.

**Comment by groenborg** [comment]
I have tried 3 libraries that should take care of suppressing terminal output when entering credentials like a password or token:  `Gopass`, `speakeasy` and `crypto/x`. Every library have the same shortcomings. When running inside a emulated terminal, they either fails to suppress the output or skips the input prompt completely. Google forum and different stack overflow channels address issues with `stty` and os specific variables. It seems to be a general problem which a lot of people have! 

- gopass issue = https://github.com/howeyc/gopass/issues/39
- speakeasy issue = https://github.com/bgentry/speakeasy/issues/23 
- I tried crypto ssh = https://stackoverflow.com/questions/2137357/getpasswd-functionality-in-go 

- thread for suppressing std in = https://stackoverflow.com/questions/38094555/golang-read-os-stdin-input-but-dont-echo-it 

I have downgraded the go version to `1.8` to see if that had any effect, unfortunately it seems like that did not solve our problem either. We run into the exactly same issues as with go version 1.9.
